File: 이제준/src/datastruct/BOJ17413.java
package datastruct;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class BOJ17413 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<Character> stack = new Stack<Character>();

        boolean tag = false;

        StringBuilder sb = new StringBuilder();
        String input = br.readLine();

        for(char ch : input.toCharArray()) {
            if(ch == '<') {
                tag = true;
                while(!stack.isEmpty()) {
                    sb.append(stack.pop());
                }
                sb.append("<");
            }else if (ch == '>') {
                tag = false;
                sb.append(">");
            }else if (ch == ' ') {
                while(!stack.isEmpty()) {
                    sb.append(stack.pop());
                }
                sb.append(" ");
            }else {
                if(!tag) stack.push(ch);
                else sb.append(ch);
            }
        }
        while(!stack.isEmpty()) {
            sb.append(stack.pop());
        }
        System.out.println(sb);
    }
}

File: 조서림/week 2/BOJ 17413.py
s = input()

tag = False
word = ""

for c in s:
    if c == "<": # tag check
        tag = True
        
    if tag == False: # tag가 아닐 때
        if c == " ":
            print(word, end=" ")
            word = ""
        else:
            word = c + word
    else: # tag일 때
        if c == ">":
            print(word, end=">")
            tag = False
            word = ""
        else:
            word += c

print(word) # 문장이 단어로 끝났을 때, 마지막 단어 출력